<launch>

	
    <group ns="ironman">  
    
    <node name="usb_cam" pkg="usb_cam" type="usb_cam_node" output="screen" >
	      <param name="video_device" value="/dev/video0" />
	      <param name="image_width" value="640" />
	      <param name="image_height" value="480" />
	      <param name="pixel_format" value="yuyv" />
	      <param name="camera_frame_id" value="usb_cam" />
	      <param name="io_method" value="mmap"/>
    </node>

	<node pkg="image_proc" type="image_proc" name="image_proc" ns="usb_cam">
	</node>
	

    <node name="input" type="input" pkg="aruco_ros">
        	<!-- <remap from="colour_image" to="usb_cam/image_rect_color" /> -->
			<remap from="colour_image" to="usb_cam/image_raw" />
            <rosparam file="$(find aruco_ros)/cfg/input.yaml" />
    </node> 


    <node pkg="aruco_ros" type="single" name="aruco_single">
        	<remap from="aruco_single/camera_info" to="usb_cam/camera_info" />
        	<remap from="aruco_single/threshold_image" to="threshold_image" />
        	<param name="image_is_rectified" value="True"/>
        	<param name="marker_size"        value="0.65"/>
        	<param name="marker_id"          value="582"/>
        	<param name="reference_frame"    value=" "/>   
        	<param name="camera_frame"       value="stereo_gazebo"/>
        	<param name="marker_frame"       value="aruco_marker_frame" />
        	<param name="corner_refinement"  value="LINES" />
    </node>

    <node name="object_pos" type="object_pos" pkg="aruco_ros">
        	<remap from="odometry" to="pilot/local_position/pose" />
        	<remap from="imu/data" to="pilot/imu/data" />
        	<remap from="aruco_single/pose" to="aruco_single/pose" />
        	<remap from="object/pose" to="object/pose" />
    </node>

    <node pkg="aruco_ros" type="filter.py" name="filter">
        	<remap from="command" to="filter/pose" />
        	<remap from="object/pose" to="object/pose" />
    </node>

    <node pkg="aruco_ros" type="fin_pos" name="fin_pos" output="screen">
    </node>
    
</group>
</launch>
